nemo/collections/nlp/data/common/sequence_to_sequence_dataset.py

```diff
@@ -146,76 +154,136 @@ def __init__(
+                )
+
+            self.samples_mapping = get_samples_mapping(
+                indexed_dataset=self.src_indexed_dataset,
+                data_prefix=self.src_file_name,
+                num_epochs=None,
+                max_num_samples=self.max_num_samples,
+                max_seq_length=self.max_src_seq_length - 2,
+                short_seq_prob=0,
+                seed=self.seed,
+                name=self.src_file_name.split('/')[-1],
+                binary_head=False,
+            )
+        else:
+            self.samples_mapping = None
+
+
+class TextMemmapSequenceToSequenceDataset(IndexedSequenceToSequenceDataset):
+    """Memory-mapped text sequence to sequence dataset. Operates on raw text files and tokenizes the text on-the-fly."""
+
+    def __init__(
+        self,
+        src_file_name: str,
+        tgt_file_name: str,
+        src_tokenizer: TokenizerSpec,
+        tgt_tokenizer: TokenizerSpec,
+        max_src_seq_length: int,
+        max_tgt_seq_length: int,
+        seed: int = 1234,
+        max_num_samples: int = None,
+    ):
+        self.seed = seed
+        self.max_num_samples = max_num_samples
+        super().__init__(
+            src_file_name=src_file_name,
+            tgt_file_name=tgt_file_name,
+            src_tokenizer=src_tokenizer,
+            tgt_tokenizer=tgt_tokenizer,
+            max_src_seq_length=max_src_seq_length,
+            max_tgt_seq_length=max_tgt_seq_length,
+            seed=seed,
+            max_num_samples=max_num_samples,
+        )
+
     def _get_examples(self):
-        self.src_dataset = TextMemMapDataset(dataset_paths=[self.src_file_name], tokenizer=self.src_tokenizer)
-        self.tgt_dataset = TextMemMapDataset(dataset_paths=[self.tgt_file_name], tokenizer=self.tgt_tokenizer)
-        assert len(self.src_dataset) == len(self.tgt_dataset), "src and tgt has different number of lines"
+        self.src_indexed_dataset = TextMemMapDataset(dataset_paths=[self.src_file_name], tokenizer=self.src_tokenizer)
+        self.tgt_indexed_dataset = TextMemMapDataset(dataset_paths=[self.tgt_file_name], tokenizer=self.tgt_tokenizer)
+
+        # Create compatibility with Megatron samples mapping
+        if self.max_num_samples is not None:
+            make_text_memmap_bin_compatibility(self.src_indexed_dataset)
+            make_text_memmap_bin_compatibility(self.tgt_indexed_dataset)
+
+        assert len(self.src_indexed_dataset) == len(
+            self.tgt_indexed_dataset
+        ), "src and tgt has different number of lines"
+        self._build_samples_mapping()
 
 
-class BinarizedMemmapSequenceToSequenceDataset(SequenceToSequenceDataset):
-    """Sequence to Sequence Dataset based on Megatron Dataset Utils."""
+class BinarizedMemmapSequenceToSequenceDataset(IndexedSequenceToSequenceDataset):
+    """Memory-mapped text sequence to sequence dataset. Operates pre-tokenized binarized data files."""
```

Comment:
Please expand on the doc string here.

Reply:
Added a docstring to the constructor here as well.